mypyc/emitclass.py

```diff
@@ -474,19 +475,36 @@ def generate_traverse_for_class(cl: ClassIR,
     for base in reversed(cl.base_mro):
         for attr, rtype in base.attributes.items():
             emitter.emit_gc_visit('self->{}'.format(emitter.attr(attr)), rtype)
+    if has_dict:
+        struct_name = cl.struct_name(emitter.names)
+        emitter.emit_gc_visit('*((PyObject **)((uintptr_t)self + sizeof({})))'.format(
```

Comment:
Document the pointer arithmetic. It's not clear why this is the right thing to do.

I'd write the pointer arithmetic using `char *` instead of `uintptr_t`, since I didn't remember what the latter does. Apparently it's been around since C99 though.

Reply:
Sure.

In a more medium term sense, I want to refactor things so that no pointer arithmetic is involved and they get fields in the structs.